### ISSUE TITLE
Sharp library causing high cpu usage on Windows

### ISSUE THREAD

**Issue by sinankeskin** [jrjohnson/ember-cli-image-transformer#117]
Hi.

First of all, thank you for this addon. Really useful.

On Windows if I delete dist folder and run ember app with ember s, nodejs using high cpu.

I investigated and found out that the reason is the Sharp library. Basically I guess Sharp process keeps open itself, never kills after success. And after the uninstall your addon high cpu usage is gone.

For the record, if I close the app (without the delete dist folder) and rerun again this issue is not happening. I guess skips the creating file.

And also this issue is not happening in Linux.

Thank you.

**Comment by jrjohnson** [comment]
Thanks for letting me know @sinankeskin. I don't have a windows machine to debug this on, but the first step is to test just sharp without this addon and see if the problem is with sharp itself or the implementation here. 

If you can extract the specifics from [generate-icons.js](https://github.com/jrjohnson/ember-cli-image-transformer/blob/master/lib/generate-icons.js) and create that reproduction it would be super helpful. If you get stuck anywhere you can ping me in Ember discord `@jrjohnson`.

**Comment by sinankeskin** [comment]
I've tested and here is the result.

Seems like this is not a Sharp library nor your implementation issue.

When I run **first time**, below line never resolve completely. Somehow hangs.
But only happening first time, not after. What I mean by that, If I restart windows and run again it's happening but after that everythings just fine.

https://github.com/jrjohnson/ember-cli-image-transformer/blob/c1b65aa9b1aca08295e3adbf3658a39adf93ac3c/lib/generate-icons.js#L26

If I change the
```javascript
return RSVP.all(promises);
```
to
```javascript
return Promise.all(promises);
```
this issue is **not** happening.

Also if I change the code like

```javascript
RSVP.all(promises).then((values) => {
  console.log(values);
}).catch(err => {
  console.log(err);
});
```

this issue is **not** happening.

Bottom line seems like this is an RSVP library issue.

Thanks for your attention. Closing.

**Comment by jrjohnson** [comment]
Looks like good sleuthing! I'd be happy to merge a PR that removes the RSVP dependency here and uses straight Promise.all which looks like it is valid for all versions of Node back to 0.12.

**Comment by sinankeskin** [comment]
Thank you.

**Comment by void-mAlex** [comment]
hello, I've upgraded to v4.0.1 but I'm still seeing the high CPU usage and I can confirm it goes away if I remove the plugin
using node v12.14.0

**Comment by sinankeskin** [comment]
Hi @void-mAlex 

Could you please test with [sharp_test](https://github.com/sinankeskin/sharp_test) and check if the node process waits or exits? Please test the same script 2 or more times.

Thanks.

**Comment by void-mAlex** [comment]
just tested and all except 1 run waited and never completed

**Comment by sinankeskin** [comment]
Hi @void-mAlex

Could you please test with [sharp_test](https://github.com/sinankeskin/sharp_test) again?

I couldn't reproduce the issue. Not in debugger nor usual run.

My node version is 12.16.2

Thanks.

**Comment by void-mAlex** [comment]
I've updated to node 12.16.2
and it still hangs on all runs and I get a bunch of warning for not being able to load fonts
sharp:12916

**Comment by sinankeskin** [comment]
I can (hopefully) confirm that this is a Sharp issue and I believe this line is the responsible.

https://github.com/lovell/sharp/blob/70e730bb677b80879d7578a4fdb881ad34afec60/lib/output.js#L767

sharp.pipeline is a native (compiled) method which unfortunately seems like on Windows it has an issue (hangs). Feel free to open an issue on Sharp library.

Also that's why I couldn't reproduce on Linux.

Sorry for the dirturb @jrjohnson and thank you @void-mAlex 

I'm closing.
